src/Proto/Semantic_JSON.hs

```diff
@@ -0,0 +1,842 @@
+-- Code generated by protoc-gen-jsonpb_haskell 0.1.0, DO NOT EDIT.
+{-# OPTIONS_GHC -Wno-orphans -Wno-unused-imports -Wno-missing-export-lists #-}
+module Proto.Semantic_JSON where
+
+import           Prelude(($), (.), (<$>), pure, show, Maybe(..))
+
+import           Data.ProtoLens.Runtime.Lens.Family2 ((^.), (.~), (&))
+import           Data.Monoid (mconcat)
+import           Control.Monad (msum)
+import           Data.ProtoLens (defMessage)
+import qualified Data.Aeson as A
+import qualified Data.Aeson.Encoding as E
+import           Data.ProtoLens.JSONPB as JSONPB
+import qualified Data.Text as T
+
+import           Proto.Semantic as P
+import           Proto.Semantic_Fields as P
+
+instance FromJSONPB PingRequest where
+  parseJSONPB = withObject "PingRequest" $ \obj -> do
+    service' <- obj .: "service"
+    pure $ defMessage
+      & P.service .~ service'
+
+instance ToJSONPB PingRequest where
+  toJSONPB x = object
+    [ "service" .= (x^.service)
+    ]
+  toEncodingPB x = pairs
+    [ "service" .= (x^.service)
+    ]
+
+instance FromJSON PingRequest where
```

Comment:
These are all orphan instances. A future alternative would be to contribute something to `proto-lens` so that this code is generated with the proto-lens-protoc plugin, but I'll leave that for another day.